views.go

```diff
@@ -391,6 +397,70 @@ func (lk *logKeeper) viewAllLogs(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (lk *logKeeper) viewTestInDatabase(r *http.Request, buildID string, testID string) (*logFetchResponse, *apiError) {
+	build, err := model.FindBuildById(buildID)
+	if err != nil || build == nil {
+		return nil, &apiError{Err: "view test by id: build not found", code: http.StatusNotFound}
+	}
+
+	test, err := model.FindTestByID(testID)
+	if err != nil || test == nil {
+		return nil, &apiError{Err: "test not found"}
+	}
+
+	logsChan, err := model.MergedTestLogs(test)
+	if err != nil {
+		lk.logErrorf(r, "Error finding global logs during test: %v", err)
+		return nil, &apiError{Err: err.Error(), code: http.StatusInternalServerError}
+	}
+
+	result := logFetchResponse{
+		logLines: logsChan,
+		build:    build,
+		test:     test,
+	}
+	return &result, nil
+}
+
+func (lk *logKeeper) viewTestInS3(r *http.Request, buildID string, testID string) (*logFetchResponse, *apiError) {
+	build, err := lk.opts.Bucket.FindBuildByID(r.Context(), buildID)
+	if err != nil || build == nil {
+		return nil, &apiError{Err: "view test in S3 by id: build not found", code: http.StatusNotFound}
+	}
+
+	test, err := lk.opts.Bucket.FindTestByID(r.Context(), buildID, testID)
+	if err != nil || test == nil {
+		return nil, &apiError{Err: "view test in S3 by id: test not found", code: http.StatusNotFound}
+	}
+
+	iterator, err := lk.opts.Bucket.GetTestLogLines(r.Context(), buildID, testID)
+	if err != nil {
+		lk.logErrorf(r, "Error finding logs during test: %v", err)
+		return nil, &apiError{Err: err.Error(), code: http.StatusInternalServerError}
+	}
+
+	logsChan := make(chan *model.LogLineItem)
+
+	go func() {
+		defer close(logsChan)
+		for iterator.Next(r.Context()) {
+			if iterator.Err() != nil {
+				lk.logErrorf(r, "Error iterating over logs: %v", err)
+				break
+			}
+			item := iterator.Item()
+			logsChan <- &item
+		}
+	}()
```

Comment:
since this is the primary place we'll be using the return of `GetTestLogLines` could it make sense for it to return a logsChan?

Reply:
Possibly... depends on if we want to add the ability to pass errors through from there. One reason the Iterator interface exists is to be able to retrieve errors using the Err() function. I'll give it a try with the channel though since we do ignore errors.

Reply:
Made a commit to use channels instead. LMK what you think. It's easy to revert if we hate it.